lib/reporters/checkstyle.js

```diff
@@ -17,12 +17,11 @@ module.exports = function(results) {
       '\'': '&apos;',
       '<': '&lt;',
       '>': '&gt;'
-    },
-    fileName, i, issue, severity, errorMessage;
+    };
 
   function encode(s) {
     for (var r in pairs) {
-      if (typeof(s) !== 'undefined') {
+      if (typeof s !== 'undefined') {
```

Comment:
While at it, this check should wrap the `for` loop, there's no point in testing `s` multiple times.